nodewatcher/modules/frontend/ip_space/static/js/draw_cidrs.js

```diff
@@ -0,0 +1,265 @@
+window.DrawCidr = class DrawCidr {
+    constructor(svg, size, subnet, url) {
+        this.svg = svg;
+        this.loadedPrefixes = 0;
+
+        this.n_ips = Math.pow(2, 32 - parseInt(subnet.split("/")[1]));
+        this.size = Math.sqrt(this.n_ips);
+        this.displaySize = this.closestPower(size);
+
+        $('#ipspace').css('transform', 'scale(' + this.displaySize / this.size + ')');
+        $('#wraper').width($('#ipspace').width() + 'px');
+        $('#wraper').height($('#ipspace').height() + 'px');
+
+        this.svg.style('width', this.size + 'px').style('height', this.size + 'px');
+
+        this.subnet = String(subnet);
+        this.ips_side = Math.sqrt(this.n_ips);
+        this.ips_pixel = this.ips_side / (this.size * 1.0);
+        this.offset = 0;
+        this.draw(subnet, "Smallest possible network to fit all the ip pools");
+        this.data = new Array(33);
+        this.base_url = url;
+        self = this;
+        $('#topnodes').on('click', '#cidr', function(event) {
```

Comment:
You are already using `class`, so you can then also use `(event) => { ... }` which will bind `this` automatically so you do not have to use `self`.

Reply:
BTW, style for anonymous functions is `function (event) {`.